src/Maestro/tests/Maestro.ScenarioTests/ScenarioTests_Subscriptions.cs

```diff
@@ -0,0 +1,272 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Maestro.ScenarioTests.ObjectHelpers;
+using Microsoft.DotNet.Darc;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Maestro.ScenarioTests
+{
+    [TestFixture]
+    [Category("ScenarioTest")]
+    public class ScenarioTests_Subscriptions : MaestroScenarioTestBase
+    {
+        private TestParameters _parameters;
+
+        private string repo1Name = "maestro-test1";
+        private string repo2Name = "maestro-test2";
+        private string repo3Name = "marstro-test3";
+        private string channel1Name = "subscriptionTestChannel1";
+        private string channel2Name = "subscriptionTestChannel2";
+
+        [SetUp]
+        public async Task InitializeAsync()
+        {
+            _parameters = await TestParameters.GetAsync();
+            SetTestParameters(_parameters);
+        }
+
+        [TearDown]
+        public Task DisposeAsync()
+        {
+            _parameters.Dispose();
+            return Task.CompletedTask;
+        }
+
+        [Test]
+        [Category("ScenarioTest")]
+        public async Task Subscriptions_EndToEnd()
```

Comment:
Given the thread in the other PR, could you file an issue to make these tests more granular? Now that we'll have them in C# we should definitely see if we can split this giant thing

Reply:
Absolutely